test/jit/test_cuda.py

```diff
@@ -49,6 +49,74 @@ def tearDown(self):
         torch.cuda.empty_cache()
         super(TestCUDA, self).tearDown()
 
+    def test_stream_args(self):
+        # Test stream creation with default arguments
+        @torch.jit.script
+        def stream_default_args() -> bool:
+            s = torch.jit.cuda.Stream()
+            return s.device_index() == torch.cuda._current_device()
+
+        @torch.jit.script
+        def stream_default_args_1() -> bool:
+            s = torch.jit.cuda.Stream(priority=0)
+            return s.device_index() == torch.cuda._current_device()
+
+        @torch.jit.script
+        def stream_default_args_2() -> bool:
+            d = torch.device("cuda:1")
+            s = torch.jit.cuda.Stream(d)
+            return s.device_index() == 1
+
+        @torch.jit.script
+        def stream_args() -> bool:
+            d = torch.device("cuda:0")
+            s = torch.jit.cuda.Stream(d, 0)
+            return s.device_index() == 0
+
+        self.assertTrue(stream_default_args)
+        self.assertTrue(stream_default_args_1)
+        self.assertTrue(stream_default_args_2)
+        self.assertTrue(stream_args)
+
+    def test_event_args(self):
+        # Test Event creation with default arguments
+        @torch.jit.script
+        def event_default_args() -> bool:
+            e = torch.jit.cuda.Event()
+            return e is not None
+
+        @torch.jit.script
+        def event_default_args_1() -> bool:
+            e = torch.jit.cuda.Event(blocking=True)
+            return e is not None
+
+        @torch.jit.script
+        def event_default_args_2() -> bool:
+            e = torch.jit.cuda.Event(enable_timing=True)
+            return e is not None
+
+        @torch.jit.script
+        def event_default_args_3() -> bool:
+            e = torch.jit.cuda.Event(interprocess=True)
+            return e is not None
+
+        @torch.jit.script
+        def event_default_args_4() -> bool:
+            e = torch.jit.cuda.Event(interprocess=True, blocking=True)
+            return e is not None
+
+        @torch.jit.script
+        def event_default_args_5() -> bool:
+            e = torch.jit.cuda.Event(enable_timing=True, blocking=True)
```

Comment:
Consider using better test names here, and testing that the arguments and default arguments are being respect. That is, the event in `event_default_args_3` should not be blocking and we should be able to verify that (I think?).

Reply:
Well, to verify blocking we would ideally need `sleep` support in JIT and since that isn't available, it becomes a lot harder to verify if blocking was truly disabled. This is the case even with interprocess. Some of the tests like `event_default_args_5` they are already tested - ref (test_event_wait) and hence I didn't add a redundant test for the same.